File: FireWall/src/main/java/com/kerolsmm/firewall/AppInfo.java
package com.kerolsmm.firewall;

import android.content.Context;
import android.content.Intent;
import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;
import android.content.pm.ResolveInfo;
import android.graphics.drawable.Drawable;

import java.util.ArrayList;
import java.util.Collections;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

public class AppInfo {
    final String label;
    final String packageName;
    final Drawable icon;
    boolean checked;

    AppInfo(String label, String packageName, Drawable icon) {
        this.label = label;
        this.packageName = packageName;
        this.icon = icon;
    }

    static List<AppInfo> getInstalledApps(Context context) {
        final Set<String> checked = context.getSharedPreferences("kerols",Context.MODE_PRIVATE)
                .getStringSet(OfflineVpnService.APPS_LIST_PREFERENCE, new HashSet<>());

        final PackageManager packageManager = context.getPackageManager();
        PackageManager pm = context.getPackageManager();
        Intent intent = new Intent(Intent.ACTION_MAIN, null);
        intent.addCategory(Intent.CATEGORY_LAUNCHER);
        List<ResolveInfo> listApps = pm.queryIntentActivities(intent, 0);
        List<AppInfo> apps = new ArrayList<>();
        for (ResolveInfo packageInfo : listApps) {
            if (packageInfo.activityInfo.packageName.equals(context.getPackageName())) continue;
            final AppInfo appInfo = new AppInfo(
                    packageInfo.activityInfo.applicationInfo.loadLabel(packageManager).toString(),
                    packageInfo.activityInfo.packageName,
                    packageInfo.activityInfo.applicationInfo.loadIcon(packageManager)
            );
            appInfo.checked = checked.contains(appInfo.packageName);
            apps.add(appInfo);
        }

        Collections.sort(apps, (a, b) -> a.label.compareTo(b.label));

        return apps;
    }

    interface FilterCallback {
        boolean filtrate(AppInfo app);
    }
}

File: Cleaner/src/main/java/com/kerols/phoneboost/Model/AppInfo.kt
package com.kerols.phoneboost.Model

import android.content.Intent
import android.content.pm.PackageManager
import android.content.pm.ResolveInfo
import android.graphics.drawable.Drawable
import android.text.format.Formatter
import androidx.fragment.app.FragmentActivity
import com.kerols.phoneboost.Activitys.MainActivity.Companion.isSystemPackage
import java.io.File


class AppInfo(app_Size: String, app_title: String?, app_icon: Drawable, pack: String) {
    var app_title: String?
    var app_Size: String
    var App_icon: Drawable
    var pack: String
    var isChecked = false

    init {
        App_icon = app_icon
        this.app_title = app_title
        this.app_Size = app_Size
        this.pack = pack
    }



    companion object {
        fun getArray(system: Boolean, context: FragmentActivity?): ArrayList<AppInfo> {
            val appInfo = ArrayList<AppInfo>()
            val pm: PackageManager? = context?.packageManager
            val intent = Intent(Intent.ACTION_MAIN, null)
            intent.addCategory(Intent.CATEGORY_LAUNCHER)
            val apps: MutableList<ResolveInfo>? = pm?.queryIntentActivities(intent, 0)
            for (app in apps ?: ArrayList()) {
                val file: File = File(app.activityInfo.applicationInfo.publicSourceDir)
                val size = file.length()
                val icon: Drawable = app.activityInfo.applicationInfo.loadIcon(pm)
                val appName = pm?.getApplicationLabel(app.activityInfo.applicationInfo).toString()
                val sizeText = Formatter.formatFileSize(context, size)
                if (!app.activityInfo.applicationInfo.packageName.equals(context?.packageName)) {
                    if (isSystemPackage(app) && system) {
                        appInfo.add(
                            AppInfo(
                                sizeText,
                                appName,
                                icon,
                                app.activityInfo.applicationInfo.packageName
                            )
                        )
                    } else if (!isSystemPackage(app) && !system) {
                        appInfo.add(
                            AppInfo(
                                sizeText,
                                appName,
                                icon,
                                app.activityInfo.applicationInfo.packageName
                            )
                        )
                    }
                }
            }
            return appInfo
        }
    }
}